CMakeLists.txt

```diff
@@ -37,8 +37,21 @@ endif()
 # Boost package
 ###############################################################################
 
+set(Boost_COMPONENTS coroutine context system thread)
+
+# If on Windows, rather than be annoyed by constant failure to find DLL errors
+# during test debugging simply force all static linking of Boost libraries
+if(WIN32 AND NOT MINGW)
+  set(Boost_USE_STATIC_LIBS ON)
+else()
+  set(Boost_COMPONENTS ${Boost_COMPONENTS} unit_test_framework)
```

Comment:
Why is `unit_test_framework` not needed by Visual Studio?

Consider adding a comment about it in the CMakeLists.txt file.